patches/api/0403-More-vanilla-friendly-methods-to-update-trades.patch

```diff
@@ -0,0 +1,67 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sun, 16 Oct 2022 15:58:38 +0200
+Subject: [PATCH] More vanilla friendly methods to update trades
+
+
+diff --git a/src/main/java/org/bukkit/entity/Villager.java b/src/main/java/org/bukkit/entity/Villager.java
+index d61855b83da0776fe910dee8cde184c720571a71..efd1f31890627204f4718cd591b31e936da08196 100644
+--- a/src/main/java/org/bukkit/entity/Villager.java
++++ b/src/main/java/org/bukkit/entity/Villager.java
+@@ -60,8 +60,11 @@ public interface Villager extends AbstractVillager {
+      * A villager with a level of 1 and no experience is liable to lose its
+      * profession.
+      *
++     * This doesn't update the trades of this villager.
++     *
+      * @param level the new level
+      * @throws IllegalArgumentException if level not between [1, 5]
++     * @see #increaseLevel(int)
+      */
+     public void setVillagerLevel(int level);
+ 
+@@ -81,6 +84,28 @@ public interface Villager extends AbstractVillager {
+     public void setVillagerExperience(int experience);
+ 
+     // Paper start
++    /**
++     * Increases the level of this villager.
++     * This also unlock new recipes unlike the raw
++     * method {@link #setVillagerLevel(int)}.
++     * A villager with a level of 1 and no experience is liable to lose its
++     * profession.
++     * A master villager has a level of 5 in its profession and
++     * will unlock 10 trades (2 per level).
++     *
++     * @param amount The amount of level
++     * @return If this villager has at least earned one level
++     * @see #setVillagerLevel(int)
++     */
++    boolean increaseLevel(int amount);
++
++    /**
++     * Gives to this villager some potential new trades
++     * based to its profession and level.
++     * @param amount The amount of trades to give
++     */
++    void addTrades(int amount);
```

Comment:
I think this should have a boolean return value. There are 2 villager professions that this method will not add trades to, the nitwit and none. The boolean would reflect if any trades were actually added.